src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1442 @@
+            if (target_mean != 0 and actual_mean != 0 and ~np.isnan(actual_mean)) else 1.0
+        scaled_lm = unscaled_lm.predict(df)
+        return scaled_lm.apply(lambda row: row * scaled_intercept)
+
+    def get_wasting_prevalence(self):
+        """ predict wasting prevalence amongst young children less than 5 years
+        :params df: population dataframe
+        """
+        # linear model to predict the prevalence of wasting
+        wasting_prevalence = LinearModel(
+            LinearModelType.LOGISTIC,
+            1.0,
+            Predictor('agrp', external=True)
+            .when('0_5mo', self.module.wasting_prevalence_odds_by_age_grp['0_5mo'])
+            .when('6_11mo', self.module.wasting_prevalence_odds_by_age_grp['6_11mo'])
+            .when('12_23mo', self.module.wasting_prevalence_odds_by_age_grp['12_23mo'])
+            .when('24_35mo', self.module.wasting_prevalence_odds_by_age_grp['24_35mo'])
+            .when('36_47mo', self.module.wasting_prevalence_odds_by_age_grp['36_47mo'])
+            .when('48_59mo', self.module.wasting_prevalence_odds_by_age_grp['48_59mo'])
+            .otherwise(1.0),
+
+            Predictor('agrp_scaling', external=True)
+            .when('0_5mo', self.module.age_grps_dict['0_5mo'])
+            .when('6_11mo', self.module.age_grps_dict['6_11mo'])
+            .when('12_23mo', self.module.age_grps_dict['12_23mo'])
+            .when('24_35mo', self.module.age_grps_dict['24_35mo'])
+            .when('36_47mo', self.module.age_grps_dict['36_47mo'])
+            .when('48_59mo', self.module.age_grps_dict['48_59mo'])
+            .otherwise(1.0),
+
+            Predictor('li_wealth').when(2, self.params['or_wasting_hhwealth_Q2'])
+            .when(3, self.params['or_wasting_hhwealth_Q3'])
+            .when(4, self.params['or_wasting_hhwealth_Q4'])
+            .when(5, self.params['or_wasting_hhwealth_Q5'])
+            .otherwise(1.0),
+            Predictor().when(
+                '(nb_size_for_gestational_age == "small_for_gestational_age") '
+                '& (nb_late_preterm == False) & (nb_early_preterm == False)',
+                self.params['or_wasting_SGA_and_term']),
+            Predictor().when(
+                '(nb_size_for_gestational_age == "small_for_gestational_age") '
+                '& (nb_late_preterm == True) | (nb_early_preterm == True)',
+                self.params['or_wasting_SGA_and_preterm']),
+            Predictor().when(
+                '(nb_size_for_gestational_age == '
+                '"average_for_gestational_age") '
+                '& (nb_late_preterm == True) | (nb_early_preterm == True)',
+                self.params['or_wasting_preterm_and_AGA'])
+        )
+
+        return wasting_prevalence
+
+    def get_moderate_acute_malnutrition_recovery(self):
+        """ predict moderate acute malnutrition recovery amongst young children less than 5 years
+        :params df: population dataframe
+        """
```

Comment:
name of function and doctoring seems different. What is returned is a model, and not a prediction, and there is no :params df: